src/families/stellar/api/horizon.js

```diff
@@ -123,9 +128,22 @@ export const fetchOperations = async (
       .cursor(startAt)
       .call();
   } catch (e) {
-    if (e instanceof NotFoundError) {
+    // FIXME: terrible hacks, because Stellar SDK fails to cast network failures to typed errors in react-native...
+    // (https://github.com/stellar/js-stellar-sdk/issues/638)
+    if (
+      e instanceof NotFoundError ||
+      (e && e.toString().match(/Error: Request failed with status code 404/g))
+    ) {
       return [];
     }
+    if (
+      e instanceof NetworkError ||
+      (e && e.toString().match(/Error: getaddrinfo ENOTFOUND/g)) ||
+      (e && e.toString().match(/undefined is not an object/g))
+    ) {
+      throw new NetworkDown();
+    }
+
```

Comment:
I don't understand this part.
Can you add a helper to "remap errors" for the whole logic ?
It should be reused everywhere there is a catch all errors that could lead to network issues.

`.match(/status code 404/)` -> no throw (return logic after the remap)

`.match(/status code 4[0-9]{2}/)` => LedgerAPI4XX

`.match(/status code 5[0-9]{2}/)` => LedgerAPI5XX

`.match(/ECONNRESET|ECONNREFUSED|ENOTFOUND|EPIPE|ETIMEDOUT/)` => NetworkDown

Reply:
There's not always a status code, sometimes it's a raw JS `undefined` error coming from the Stellar SDK...

Reply:
But I can replace this one `.match(/Error: getaddrinfo ENOTFOUND/g))` with the more detailed cases you mention.

Reply:
yeah ok, so we can just add some cases. Strange behaviour though. Idea is just to handle as much (known) cases as possible.